File: Dataset/Concurrency_prepared/Python/CountEvenOdd_trace_1.py
def count_even(numbers, result):
    count = sum(1 for num in numbers if num % 2 == 0)
    result.append(count)
    print(f"Count of even numbers: {count}")

# Function to count odd numbers
def count_odd(numbers, result):
    count = sum(1 for num in numbers if num % 2 != 0)
    result.append(count)
    print(f"Count of odd numbers: {count}")

if __name__ == "__main__":
    numbers = [11, 22, 33, 44, 55, 66, 77, 88, 99, 100]
    result = []

    even_thread = threading.Thread(target=count_even, args=(numbers, result))
    odd_thread = threading.Thread(target=count_odd, args=(numbers, result))

    even_thread.start()
    #START
    odd_thread.start()

    even_thread.join()
    odd_thread.join()
    #END

    total_even, total_odd = result
    print(f"Total even: {total_even}, Total odd: {total_odd}")

File: Dataset/Advanced/Test/Concurrency_prepared/Java/CountEvenOdd_trace_1.java
import java.util.Arrays;

class CountEven implements Runnable {
    private int[] numbers;
    private int[] result;

    public CountEven(int[] numbers, int[] result) {
        this.numbers = numbers;
        this.result = result;
    }

    @Override
    public void run() {
        int count = (int) Arrays.stream(numbers).filter(num -> num % 2 == 0).count();
        result[0] = count;
        System.out.println("Count of even numbers: " + count);
    }
}

class CountOdd implements Runnable {
    private int[] numbers;
    private int[] result;

    public CountOdd(int[] numbers, int[] result) {
        this.numbers = numbers;
        this.result = result;
    }

    @Override
    public void run() {
        int count = (int) Arrays.stream(numbers).filter(num -> num % 2 != 0).count();
        result[1] = count;
        System.out.println("Count of odd numbers: " + count);
    }
}

public class ConcurrencyExample {
    public static void main(String[] args) throws InterruptedException {
        int[] numbers = {11, 22, 33, 44, 55, 66, 77, 88, 99, 100};
        int[] result = new int[2];

        Thread evenThread = new Thread(new CountEven(numbers, result));
        Thread oddThread = new Thread(new CountOdd(numbers, result));

        evenThread.start();
        oddThread.start();

        evenThread.join();
        oddThread.join();

        System.out.println("Total even: " + result[0] + ", Total odd: " + result[1]);
    }
}